### PR TITLE
bump kube-openapi

### PR DESCRIPTION
**Pull request by liggitt**
/assign @apelisse 
/kind bug

Picks up:
- [x] https://github.com/kubernetes/kube-openapi/pull/347
- [x] https://github.com/kubernetes/kube-openapi/pull/349

In the process of updating here, discovered and resolved these issues as well:
- [x] https://github.com/kubernetes/kube-openapi/pull/353
- [x] https://github.com/kubernetes/kube-openapi/pull/352
- [x] https://github.com/kubernetes/kube-openapi/pull/351

```release-note
Fixes panic validating custom resource definition schemas that set `multipleOf` to 0
```

### CODE REVIEW

**Comment by apelisse** [comment]
There are actually quite a few changes but nothing I think is specifically critical. Thanks.
/lgtm

**Comment by pacoxu** [comment]
/hold 
for CI failures
```
2023/01/05 23:43:29 OpenAPI code generation error: Failed executing generator: some packages had errors:
errors in package "k8s.io/kubernetes/pkg/generated/openapi":
output for "openapi/zz_generated.openapi.go" differs; first existing/expected diff: 
  "\",\n\t\t\t\t\t\t\tType:        []string{\"string\"},\n\t\t\t\t\t\t\tFormat:      \"\",\n\t\t\t\t\t\t},\n\t\t\t\t\t},\n\t\t\t\t\t\"matchPolic"
  "\\n\\nPossible enum values:\\n - `\\\"Fail\\\"` means that an error calling the webhook causes the admissio"

!!! [0105 23:43:29] Call tree:
!!! [0105 23:43:29]  1: /home/prow/go/src/k8s.io/kubernetes/hack/update-codegen.sh:836 codegen::openapi(...)
+++ exit code: 1
+++ error: 1
[0;31mFAILED[0m   verify-codegen.sh	147s
V
```

**Comment by liggitt** [comment]
@apelisse seeing error serving OpenAPI:

> Error from server (InternalError): Internal error occurred: mime: unexpected content after media subtype

File: vendor/k8s.io/kube-openapi/pkg/handler/handler.go [hunk 1]
```diff
@@ -178,8 +179,8 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 							return
 						}
 					}
-					contentType := accepts.Type + "/" + accepts.SubType
-					w.Header().Set("Content-Type", contentType)
+					// contentType := accepts.Type + "/" + accepts.SubType
+					// w.Header().Set("Content-Type", contentType)
```
Comment (liggitt) [inline vendor/k8s.io/kube-openapi/pkg/handler/handler.go:183]: @apelisse @Jefftree commenting this out made tests pass again... something about the content type being set is making the server choke. I think it does not like the @ symbol in the proto subtype, based on the error message coming from here:

https://cs.opensource.google/go/go/+/release-branch.go1.19:src/mime/mediatype.go;l=103-122

https://cs.opensource.google/go/go/+/release-branch.go1.19:src/mime/grammar.go;l=11-32;drc=119f679a3bd2e60cfc990920f82fd1a5cb006f4c

`@` seems to be disallowed as a token character by that library

**Comment by jiahuif** [comment]
Do we backport it for https://github.com/kubernetes/kube-openapi/pull/349 ?

**Comment by apelisse** [comment]
Should we backport only that change? We have the option of making a branch in kube-openapi for the specific release and just release that change if needed.

**Pull request merged**